docker/auth.go

```diff
@@ -70,7 +75,13 @@ func (ba BasicAuthConfig) Empty() bool {
 }
 
 // Validate validates the spec.
-func (ba BasicAuthConfig) Validate() error {
+func (ba *BasicAuthConfig) Validate() error {
+	if ba.Empty() {
+		return nil
+	}
+	if ba.Auth == "" {
+		ba.Auth = base64.StdEncoding.EncodeToString([]byte(ba.Username + ":" + ba.Password))
+	}
```

Comment:
Validate() shouldn't have a side effect. Can we set Auth elsewhere, eg when the struct is created or something